File: 2078. Two Furthest Houses With Different Colors/solution.java
class Solution {
  public int maxDistance(int[] colors) {
    int furthestFromStart = colors.length - 1;
    int furthestFromEnd = 0;
    while (colors[furthestFromStart] == colors[0]){
      furthestFromStart -= 1;
    }
    while (colors[furthestFromEnd] == colors[colors.length - 1]){
      furthestFromEnd += 1;
    }       
    
    int max = colors.length - 1 - furthestFromEnd;
    if (max < furthestFromStart){
      max = furthestFromStart;
    }
    
    return max;
  }
}

File: 2078. Two Furthest Houses With Different Colors/solution.py
class Solution(object):
  def maxDistance(self, colors):
    """
    :type colors: List[int]
    :rtype: int
    """
    furthestFromStart = len(colors) - 1
    furthestFromEnd = 0
    while colors[furthestFromStart] == colors[0]:
      furthestFromStart -= 1
    while colors[furthestFromEnd] == colors[-1]:
      furthestFromEnd += 1
    
    return max(len(colors) - 1 - furthestFromEnd, furthestFromStart)